src/api/app/models/staging_workflow.rb

```diff
@@ -21,11 +21,7 @@ def ready_to_stage
   after_create :create_staging_projects
 
   def unassigned_requests
-    target_of_bs_requests.stageable - staged_requests - ignored_requests
-  end
-
-  def ready_requests
-    target_of_bs_requests.ready_to_stage - staged_requests - ignored_requests
+    target_of_bs_requests.in_states(['new', 'review']).where.not(id: staged_requests.pluck(:id)).where.not(id: ignored_requests.pluck(:id))
```

Comment:
drop the `pluck`s! :see_no_evil:

You can write this more elegantly and simply:

```ruby
target_of_bs_requests.in_states(['new', 'review']).where.not(id: staged_requests | ignored_requests)
```

:bowtie: 

or even more:

```ruby
target_of_bs_requests.in_states(['new', 'review']) - (staged_requests | ignored_requests)
```
or:

```ruby
target_of_bs_requests.in_states(['new', 'review']) - staged_requests - ignored_requests
```

:bowtie: :bowtie:

Reply:
The last two do not work as they transform the results into an array, this is not what we want (see the diff, this is how it was before)

Reply:
ok, then please use the first one :wink: